src/Lachain.Core/RPC/HTTP/AccountService.cs

```diff
@@ -120,7 +122,7 @@ private JObject DeployContract(string byteCodeInHex, string input, ulong gasLimi
                 throw new ArgumentException("Unable to validate smart-contract code");
             // TODO: use deploy abi if required
             var tx = _transactionBuilder.DeployTransaction(from, byteCode);
-            var signedTx = _transactionSigner.Sign(tx, ecdsaKeyPair);
+            var signedTx = _transactionSigner.Sign(tx, ecdsaKeyPair, HardforkHeights.IsHardfork_8Active(_stateManager.LastApprovedSnapshot.Blocks.GetTotalBlockHeight()));
```

Comment:
If this tx is later added to the pool, then we should use GetTotalBlockHeight() + 1